### PR TITLE
GREP-xxxx: Pull GRC out of tree

### PR DESCRIPTION
**Pull request by mormj**
A proposal to make GRC more modular so it could suit the needs of projects beyond in-tree GNU Radio

Signed-off-by: Josh Morman <jmorman@gnuradio.org>

### CODE REVIEW

**Comment by 777arc** [comment]
Might as well also take the reverse perspective and see if there's any flowgraph-style FOSS that can be extended (without too much code) to do what we need it to do for GR, for the sake of code reuse and immediate "partners" who use the tool for other purposes.  Like does Octave have a Simulink style interface?  I have no idea if anything exists but it's worth compiling a list somewhere.  Perhaps at the end of the GREP we can have a list of the most similar open source software to GRC.

**Comment by willcode** [comment]
@777arc  Worth looking just to see what's out there any maybe get ideas. Some "close but no cigar" things I found include Dia and GNS3. The closest thing in the SDR domain is PothosFlow, which is tied to the underlying Pothos architecture. There's some decent overlap with block-based coding (Blockly, Scratch, Codejig) since it has a back-end of some sort.

**Comment by 777arc** [comment]
What about Xcos?  https://www.scilab.org/software/xcos  
EDIT- looks like the code is a module within scilab's repo, so that doesn't help

**Comment by mormj** [comment]
I'm intrigued by blockly - perhaps would allow for us to solve things like: https://github.com/gnuradio/gnuradio/issues/5097 in the process of creating grc flowgraphs.

File: grep-9999-grc-out-of-tree.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
+# GREP [XXXX] -- [Move GRC Out of Gree]
```
Comment (marcusmueller) [inline grep-9999-grc-out-of-tree.md:1]: GREE -> GR Tree?

**Review by marcusmueller — APPROVED** [review]

File: grep-9999-grc-out-of-tree.md (hunk 1 above)
Comment (mormj) [reply grep-9999-grc-out-of-tree.md:1]: it's a convenient abbreviation :man_facepalming:

**Comment by marcusmueller** [comment]
@mormj  I think this GREP is acceptable! Is it OK if I fix that typo, assign a number and merge it?

**Comment by mormj** [comment]
Agreed - feel free to go ahead!  Thanks!

**Comment by marcusmueller** [comment]
Manually merged

**Pull request closed without merging**